### ISSUE TITLE
Salt-cloud (Carbon) does not create security groups in EC2

### ISSUE THREAD

**Issue by rehmanzile** [saltstack/salt#38286]
### Description of Issue/Question
Salt-cloud (Carbon) does not create security groups in EC2

### Setup
Create the /etc/salt/cloud/profiles.d/my_profile.conf and paste the following:
```
#/etc/salt/cloud/profiles.d/my_profile.conf
ec2_west_c4_2xlarge_manager_spot_onprem:
  provider: ec2-us-west-2-manager-onprem-private
  ssh_interface: private_ips
  image: ami-01f05461
  size: m3.2xlarge
  spot_config:
    spot_price: 5
  ssh_username: poncho
  environment: onprem
  minion:
    grains:
        role: manager
        environment: onprem
  tag: {'Environment': 'onprem'}
  sync_after_install: grains
```

Create the /etc/salt/cloud.providers.d/my_provider.conf and paste the following:
```
ec2-us-west-2-manager-onprem-private:
  minion:
    master: onprem-provisioner.internal.nyansa.com
  id: 'XXXXXX'
  key: 'XXXXXXXXXXX'
  private_key: /etc/salt/XXXXX.pem
  keyname: XXXXX
  ssh_interface: private_ips
  subnetid: subnet-55af7431
  securitygroupid:
    - sg-38ecc95e
    - sg-a5bb5ac1
    - sg-50d63034
    - sg-3870ae41
  location: us-west-2
  availability_zone: us-west-2a
  driver: ec2
  rename_on_destroy: True
```

### Steps to Reproduce Issue
Try spinning up the instance in Amazon EC2.
```
sudo salt-cloud -y -p ec2_west_c4_2xlarge_manager_spot_onprem onprem-manager01
```

You will notice the SECURITY GROUPS are still set to default. 

### Versions Report
$ salt-cloud --version
salt-cloud 2016.11.0 (Carbon)

**Comment by Ch3LL** [comment]
@rehmanzile can you give the fix in #38183 a try please?

**Comment by rehmanzile** [comment]
@Ch3LL Thanks. Do you have instructions on how to I can pick up the fix?
